### PR TITLE
msg/async: drop unused member variable

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
this changeset will conflict with #38415. i will rebase mine once the latter gets merged.

File: src/msg/async/Event.h [hunk 1]
```diff
@@ -153,7 +153,7 @@ class EventCenter {
 
  private:
   CephContext *cct;
-  std::string type;
+  std::string_view type;
```
Comment (batrick) [inline src/msg/async/Event.h:156]: This deserves a comment on how we prevent dangling string references. I'm not even sure it's not possible...

File: src/msg/async/Stack.cc [hunk 2]
```diff
@@ -102,7 +102,8 @@ Worker* NetworkStack::create_worker(CephContext *c, const std::string &type, uns
   return nullptr;
 }
 
-NetworkStack::NetworkStack(CephContext *c, const std:: string &t): type(t), started(false), cct(c)
+NetworkStack::NetworkStack(CephContext *c, std:: string_view t)
```
Comment (batrick) [inline src/msg/async/Stack.cc:105]:
```suggestion
NetworkStack::NetworkStack(CephContext *c, std::string_view t)
```

**Review by batrick — CHANGES_REQUESTED** [review]

File: src/msg/async/Event.h (hunk 1 above)
Comment (tchaikov) [reply src/msg/async/Event.h:156]: yeah, i should have done this -- this `type` originally is passed down as a string literal. but after a second thought, i think it's not worthwhile to trade the couple bytes with the readability. i will drop this commit.

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-12-06_12:59:57-rados-wip-kefu-testing-2020-12-06-1640-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-12-08_08:10:54-rados-wip-kefu-testing-2020-12-06-1640-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-12-09_01:18:38-rados-wip-kefu-testing-2020-12-08-2025-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-12-09_08:36:52-rados-wip-kefu-testing-2020-12-08-2025-distro-basic-smithi

failures are not related.

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-12-10_14:22:12-rados-wip-kefu-testing-2020-12-10-1827-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-12-11_07:09:12-rados-wip-kefu-testing-2020-12-10-1827-distro-basic-smithi/

failures are not related.

**Comment by tchaikov** [comment]
@batrick could you please take another look?

**Review by batrick — APPROVED** [review]

**Pull request merged**